codes/javascript/chapter_array_and_linkedlist/array.js

```diff
@@ -0,0 +1,95 @@
+// File: array.js
+// Created Time: 2022-11-27
+// Author: IsChristina (christinaxia77@foxmail.com)
+
+// 随机访问元素
+function randomAccess(nums){
+    // 在区间 [0, nums.length) 中随机抽取一个数字
+    const random_index  = Math.floor(Math.random() * nums.length)
+    // 获取并返回随机元素
+    random_num = nums[random_index]
+    return random_num
+}
+// 扩展数组长度 js语言中的数组长度是动态的 一般无需指定长度
+function extend(nums, enlarge){
+   // 初始化一个扩展长度后的数组
+   let res = new Array(nums.length + enlarge).fill(0)
+   // 将原数组中的所有元素复制到新数组
+   for(let i=0; i<nums.length;i++){
+        res[i] = nums[i]
+   }
+    // 返回扩展后的新数组
+    return res
+}
+// 在数组的索引 index 处插入元素 num
+function insert(nums, num, index){
+    // 把索引 index 以及之后的所有元素向后移动一位
+    for (let i = nums.length - 1; i >= index; i--) {
+        nums[i] = nums[i - 1];
+    }
+    // 将 num 赋给 index 处元素
+    nums[index] = num;
+}
+
+// 删除索引 index 处元素
+function remove(nums, index){
+    // 把索引 index 之后的所有元素向前移动一位
+    for (let i = index; i < nums.length - 1; i++) {
+        nums[i] = nums[i + 1]
+     }
+}
+    
+// 遍历数组 
+function traverse(nums){
+    let count = 0
+    // 通过索引遍历数组
+    for (let i = 0; i < nums.length; i++) {
+       count++;
+    }
+    // 直接遍历数组
+    for(let num of nums){
+         count += 1
+     }
+}
+
+// 在数组中查找指定元素
+function find(nums, target){
+    for (let i = 0; i < nums.length; i++) {
+        if (nums[i] == target)
+            return i;
+    }
+    return -1
+}
+
+// 初始化数组
+var arr = new Array(5).fill(0)
+console.log("数组 arr =", arr)
+var nums = [1, 3, 2, 5, 4]
+console.log("数组 nums =", nums)
+
+// 随机访问
+random_num = randomAccess(nums)
+console.log("在 nums 中获取随机元素", random_num)
+
+// 长度扩展
+nums = extend(nums, 3)
+console.log("将数组长度扩展至 8 ，得到 nums =", nums)
+
+// 插入元素
+insert(nums, 6, 3)
+// 还可以直接使用ES6 splice
```

Comment:
This should be introduced in the section "list".
In this section, we assume that it is an array (although it is a dynamic array)